src/traces/scattermapbox/convert.js

```diff
@@ -32,19 +32,39 @@ module.exports = function convert(gd, calcTrace) {
     var hasLines = subTypes.hasLines(trace);
     var hasMarkers = subTypes.hasMarkers(trace);
     var hasText = subTypes.hasText(trace);
-    var hasCircles = (hasMarkers && trace.marker.symbol === 'circle');
-    var hasSymbols = (hasMarkers && trace.marker.symbol !== 'circle');
+    var hasCircles = hasMarkers && trace.marker.symbol === 'circle';
+    var hasSymbols = hasMarkers && trace.marker.symbol !== 'circle';
+    var hasCluster = trace.cluster.enabled;
 
-    var fill = initContainer();
-    var line = initContainer();
-    var circle = initContainer();
-    var symbol = initContainer();
+    var fill = initContainer('fill');
+    var line = initContainer('line');
+    var circle = initContainer('circle');
+    var symbol = initContainer('symbol');
 
     var opts = {
         fill: fill,
         line: line,
         circle: circle,
-        symbol: symbol
+        symbol: symbol,
+        cluster: {
```

Comment:
We should possibly move these `cluster` and `clusterCount` below and put them inside `if(hasCircles) {` block.